flake.nix

```diff
@@ -204,9 +207,8 @@
 
           # Build *just* the cargo dependencies, so we can reuse all of that work (e.g. via cachix) when running in CI
           cargoArtifacts = craneLib.buildDepsOnly {
-            pname = "nickel-lang";
-            inherit src;
-            cargoExtraArgs = "${cargoBuildExtraArgs} --workspace";
+            inherit pname src;
+            cargoExtraArgs = "${cargoBuildExtraArgs} --workspace --all-features";
```

Comment:
Why do we need to specify `--all-features` when building the dependencies?

Reply:
Because the `doc` build uses `--all-features`, so without it the doc build will rebuild all the dependencies. (at least i think that's what was happening, and this change seemed to fix that). 

Does it cause any problems?

Reply:
I don't think there's any issue, I was just curious. Thanks for the clarification!